pkg/cmd/gist/edit/edit_test.go

```diff
@@ -100,6 +119,10 @@ func TestNewCmdEdit(t *testing.T) {
 			cmd.SetErr(&bytes.Buffer{})
 
 			_, err = cmd.ExecuteC()
+			if tt.wantsErr {
+				assert.Error(t, err)
```

Comment:
This is not a required change but just FYI for future it's nice to use `require.Error(t, err)` here and then remove the `return` on the next line.

So many times I've seen a missed `return` and the test output ends up making no sense. In fact I would argue that `assert.Error(t, err)` is pretty much never the right choice since doing anything after it other than exiting means your test is in a very unexpected state.

Reply:
I don't think that's true, `require` should call `FailNow` causing immediate exit of the test, that's the difference between `require` and `assert`.

Reply:
yes right, but that's when the testcase fails.

The logic is, if the testcase is expecting an error `wantsErr = true` make sure the err variable is not nil and then return cause no point checking the other assertions, otherwise `wantsErr = false` check the other assertions

Reply:
😅 I think there is some confusion here and we're talking past each other but it's ok, we'll get to the bottom of it! Possibly because I said `require.NoError` when I meant `require.Error`

We're both saying:

> We should check that if `wantsErr = true` then fail the test and exit.

I think you're saying that we need `return` after `require.Error(t, err)` to ensure that we exit and don't run further assertions.

I'm saying that `require.Error(t, err)` causes immediate exit of the test when it fails. We will never proceed to any other line.

See the following example: https://go.dev/play/p/ZQRuYjI-sgu

Reply:
Sorry, I understand what you're saying now. If `wantsErr` is set to true in the test setup, we want to early exit the test either way because the rest of the assertions don't make sense. Gotcha!

Reply:
No worries 🤘